spec/rubocop/cop/rspec/factory_bot/factory_class_name_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::FactoryBot::FactoryClassName do
+  subject(:cop) { described_class.new }
+
+  context 'when passing block' do
+    it 'registers an offense when using passing class' do
+      expect_offense(<<~RUBY)
+        factory :foo, class: Foo do
+                             ^^^ Pass 'Foo' instead of Foo.
```

Comment:
Just in case, do you mind adding a check, or just make sure `::Foo` and `Foo::Bar` are also flagged?